### PR TITLE
feat: Add Collapsing Message

### PR DESCRIPTION
**Pull request by TheShubham99**
#### Behaviour:
Read more button appears when message length goes beyond certain number of chars.
This length can be modified in Constants.jsx

Link - http://theshubham99.github.io/AdventureSite

This will be consistent across all kinds of messages. (from user & gooey) and a separate component can be built if needed in the future.

@pollend @smsunarto  @majordwarf

### CODE REVIEW

File: src/components/UserMessageSection.jsx [hunk 1]
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 
 export function UserMessageSection() {
   return (
-    <div id='user-message-section'>
+    <div className='user-message-section'>
       <p style={{ color: '#fff', fontSize: '200%' }}>
```
Comment (majordwarf) [inline src/components/UserMessageSection.jsx:6]: I guess the following inline CSS will be removed once the design is finalized? :thinking:

File: src/styles/index.css [hunk 2]
```diff
@@ -99,15 +99,25 @@ body {
...
+  padding: 2px;
+  margin: 2px;
+  height: auto;
   background: rgba(0, 0, 0, 0.8);
   border: solid 3px #fff;
   margin-bottom: 500px;
 }
 
+.message-collapse-button {
+  padding: 1;
```
Comment (majordwarf) [inline src/styles/index.css:114]: Invalid CSS property, the value of padding should have some metric (px, em, rem etc.).
Fix the following thing, amend the commit and force push the branch, no need to create a new commit.

File: src/styles/index.css [hunk 3]
```diff
@@ -99,15 +99,25 @@ body {
...
+  color: yellow;
```
Comment (majordwarf) [inline src/styles/index.css:115]: Using SCSS would be much better so we can have common CSS files for all of our colours, common styles, typography.

File: src/Constants.jsx [hunk 4]
```diff
@@ -0,0 +1,6 @@
+export const Constants = {
+  readMore: '... Read More',
+  showLess: 'Show Less',
+  // max chars to be shown in collapsed view of the message
+  maxChars: 130,
+};
```
Comment (majordwarf) [inline src/Constants.jsx:6]: Always leave a newline at end of file, it's a good practice. You can read more about that [here](https://stackoverflow.com/questions/2287967/why-is-it-recommended-to-have-empty-line-in-the-end-of-a-source-file#:~:text=The%20empty%20line%20in%20the,can%20handle%20the%20EOF%20marker.)

File: src/components/UserMessageSection.jsx (hunk 1 above)
Comment (TheShubham99) [reply src/components/UserMessageSection.jsx:6]: Yes. It's placeholder. I am currently working on it.

File: src/styles/index.css (hunk 3 above)
Comment (TheShubham99) [reply src/styles/index.css:115]: Sure. will look into that before next PR. :)

File: src/Constants.jsx (hunk 4 above)
Comment (TheShubham99) [reply src/Constants.jsx:6]: Sure. May I use it from next time ?(PR) as it's conflicting with few other branches. @majordwarf

**Comment by TheShubham99** [comment]
> The commit should have space after `feat:`, so the final commit message would look like this,
> 
> ```
> feat: Added Collapsing Message
> ```
> 
> Also, try to use imperative words i.e. (Add, Fix, Delete) instead of `Added` and according to conventional commits the message should be in the lower case until and unless using names of some particular file. So, a better message would be `feat: add collapsable message feature to gooey's chat.`

#### @majordwarf Done 👍

File: src/Constants.jsx (hunk 4 above)
Comment (pollend) [reply src/Constants.jsx:6]: not that much additional work to just fix the change then and there.

**Pull request merged**